core/src/commonMain/kotlin/komem/litmus/LitmusRunner.kt

```diff
@@ -60,46 +60,39 @@ abstract class LitmusRunner {
         return allJoinHandles
     }
 
-    // 1) be extremely careful due to LitmusOutcome = Any?
-    // 2) Sequence forces to think twice before accidentally allocating a huge list (like `states.map{...}`)
-    // 3) this is an optimized version which is ~30% faster than `.groupingBy().eachCount()` on List-s (!)
-    protected fun Sequence<LitmusOutcome>.calcStats(outcomeSpec: LitmusOutcomeSpec): LitmusResult {
-        val bucketsMap = mutableMapOf<Int, Pair<MutableList<LitmusOutcome>, MutableList<Long>>>()
+    protected fun <S : Any> calcStats(
+        states: List<S>,
+        spec: LitmusOutcomeSpec,
+        outcomeFinalizer: (S) -> LitmusOutcome
+    ): LitmusResult {
+        // cannot do `map.getOrPut(key){0L}++` with Long-s, and by getting rid of one
+        // extra put(), we are also getting rid of one extra hashCode()
+        class LongHolder(var value: Long)
 
-        fun fastEquals(o1: Any?, o2: Any?): Boolean {
-            if (o1 is List<*> && o2 is List<*>) {
-                if (o1.size != o2.size) return false
-                for (i in o1.indices) {
-                    if (o1[i] != o2[i]) return false
-                }
-                return true
-            } else return o1 == o2
-        }
+        // the absolute majority of outcomes will be declared in spec
+        val fastPathOutcomes = (spec.accepted + spec.interesting + spec.forbidden).toCustomList()
```

Comment:
maybe rename to `specifiedOutcomes` ?
(since the name `fastPathOutcomes` refers to an optimization being used)

Reply:
ok